pkg/build/api/types.go

```diff
@@ -425,6 +428,19 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// JenkinsPipelineStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
```

Comment:
My only concern with that naming is if Jenkins Pipeline introduces more sources than just the Jenkinsfile. Would you then introduce more build strategies for each possible Pipeline source? I see Jenkins Pipeline as the strategy: that is the build process, Jenkinsfile is the input to that strategy.